pgmpy/inference/ApproxInference.py

```diff
@@ -71,14 +80,18 @@ def get_distribution(self, samples, variables, state_names=None, joint=True):
         if isinstance(variables, (set, tuple)):
             variables = list(variables)
 
+        # Get state names from the model
+        model_states = {var: self.model.states[var] for var in variables}
```

Comment:
Isn't `model_states` equivalent to `self.model.states` here? Are you trying to make a copy of it?